### PR TITLE
upgrade date-fns to v2.0.1

### PR DESCRIPTION
**Pull request by bengry**
[`date-fns@2` has been released](https://github.com/date-fns/date-fns/releases/tag/v2.0.0), and since `concurrently` is pinned on `^1.x.x`, it makes it hard for users `concurrently` to use `date-fns@2` simultaneously, due to conflicting packages.

This PR updates this dependency.

Note that this is a breaking change in `concurrently` for users who may have passed an invalid format (which previously worked) due to [a breaking change in `date-fns@2.0.0`, to better conform the unicode standard](https://github.com/date-fns/date-fns/blob/master/docs/unicodeTokens.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,11 +10,13 @@ Like `npm run watch-js & npm run watch-less` but better.
...
-- [Usage](#usage)
-- [Programmatic Usage](#programmatic-usage)
-- [FAQ](#faq)
+- [Concurrently](#concurrently)
+  - [Why](#why)
+  - [Install](#install)
+  - [Usage](#usage)
+  - [Programmatic Usage](#programmatic-usage)
+    - [`concurrently(commands[, options])`](#concurrentlycommands-options)
+  - [FAQ](#faq)
```
Comment (gustavohenke) [inline README.md]: Can you please undo the changes to this TOC?

Comment (bengry) [reply README.md]: Yea, I didn't notice it at first. Must've been done by one of my VS Code extensions. Will revert.

**Comment by bengry** [comment]
@gustavohenke technically yes, though it should only users of the progrematic API, and even then - only those who override the default timestamp format.

**Comment by bengry** [comment]
@gustavohenke / @kimmobrunfeldt can this PR be merged, or is it waiting for something? It was approved almost a month ago.

**Comment by gustavohenke** [comment]
Hey everyone! Sorry for the delay. I was waiting for the approval to turn 1 month old 🎂 

Jokes apart... you may start using this on [`v5.0.0`](https://github.com/kimmobrunfeldt/concurrently/releases/tag/v5.0.0) 🚢
